central/declarativeconfig/updater/notifier_updater.go

```diff
@@ -83,24 +88,27 @@ func (u *notifierUpdater) DeleteResources(ctx context.Context, resourceIDsToSkip
 		// while integration health isn't.
 		if err := u.notifierDS.RemoveNotifier(ctx, n.GetId()); err != nil && !errors.Is(err, errox.NotFound) {
 			err := errors.Wrap(err, "deleting notifier from database")
-			notifierDeletionErr, notifierIDs = u.processDeletionError(notifierDeletionErr, err, notifierIDs, n)
+			notifierDeletionErr, notifierIDs = u.processDeletionError(ctx, notifierDeletionErr, err, notifierIDs, n)
 			continue
 		}
 
 		u.processor.RemoveNotifier(ctx, n.GetId())
 		if err := u.reporter.RemoveIntegrationHealth(n.GetId()); err != nil {
 			err := errors.Wrap(err, "deleting notifier's integration health")
-			notifierDeletionErr, notifierIDs = u.processDeletionError(notifierDeletionErr, err, notifierIDs, n)
+			notifierDeletionErr, notifierIDs = u.processDeletionError(ctx, notifierDeletionErr, err, notifierIDs, n)
 		}
 	}
 	return notifierIDs, notifierDeletionErr.ErrorOrNil()
 }
 
-func (u *notifierUpdater) processDeletionError(notifierDeletionErr *multierror.Error, err error, notifierIDs []string, notifier *storage.Notifier) (*multierror.Error, []string) {
+func (u *notifierUpdater) processDeletionError(ctx context.Context, notifierDeletionErr *multierror.Error, err error,
+	notifierIDs []string, notifier *storage.Notifier) (*multierror.Error, []string) {
 	notifierDeletionErr = multierror.Append(notifierDeletionErr, err)
 	notifierIDs = append(notifierIDs, notifier.GetId())
 
-	u.reporter.UpdateIntegrationHealthAsync(declarativeCfgUtils.IntegrationHealthForProtoMessage(notifier, "", err,
-		u.idExtractor, u.nameExtractor))
+	if err := u.healthDS.UpsertDeclarativeConfig(ctx, declarativeCfgUtils.HealthStatusForProtoMessage(notifier, "", err,
```

Comment:
It's sad that `handler` value goes missing here even if it was set before. Perhaps in a follow-up instead of upsert we should do get+merge for config health?

Reply:
Yeah seems like we lived with it previously since it isn't really changed compared to beforehand.

I can create a follow-up PR to come up with something to address that (maybe just a helper that only updates error message + timestamp is the best thing).